hbase-server/src/main/java/org/apache/hadoop/hbase/master/ActiveMasterManager.java

```diff
@@ -57,12 +56,18 @@
   final AtomicBoolean clusterHasActiveMaster = new AtomicBoolean(false);
   final AtomicBoolean clusterShutDown = new AtomicBoolean(false);
 
+  // This server's information.
   private final ServerName sn;
   private int infoPort;
   private final Server master;
 
+  // Active master's server name. Invalidated anytime active master changes (based on ZK
+  // notifications) and lazily fetched on-demand.
+  // ServerName is immutable, so we don't need heavy synchronization around it.
```

Comment:
Actually, I think this field can replace the `AtomicBoolean clusterHasActiveMaster`. Seems your new information is a super-set of the existing information.

Reply:
No, this is a fair point. It would be nice to consolidate these concerns, but if we don't have the test coverage and you don't have the capacity for a manual test, better leave it as is.